client/my-sites/invites/controller.js

```diff
@@ -50,10 +49,10 @@ export function acceptInvite( context, next ) {
 				const redirect = getRedirectAfterAccept( acceptedInvite );
 				debug( 'Accepted invite and redirecting to:  ' + redirect );
 
-				if ( get( acceptedInvite, 'site.is_wpforteams_site', false ) ) {
-					// Using page() here will throw an error because P2 sites
-					// redirect to non-same origin.
-					window.location.href = redirect;
+				// Using page() for cross origin navigations would throw a `History.pushState` exception
+				// about creating state object with a cross-origin URL.
+				if ( new URL( redirect ).origin !== window.location.origin ) {
```

Comment:
`redirect` uses `getRedirectAfterAccept()`, and it seems like under certain circumstances, it might [return a path](https://github.com/Automattic/wp-calypso/blob/a126f4799fea0a79283084594be715bd91293c8b/client/my-sites/invites/utils.js#L178). Won't `new URL( '/some/path' )` trigger a `TypeError`?

Reply:
Good catch! I added a second param (`base`) to the `new URL( url, base )` calls so that relative URLs have something to resolve against. In 138ba9f.